.github/workflows/erc-registry-indexer.yml

```diff
@@ -0,0 +1,162 @@
+name: ERC Registry Indexer
+
+on:
+  workflow_dispatch:
+    inputs:
+      HEDERA_NETWORK:
+        description: 'Target Hedera Network (e.g., previewnet, testnet, mainnet).'
+        required: true
+      MIRROR_NODE_URL:
+        description: 'Mirror Node Base URL (e.g., https://testnet.mirrornode.hedera.com).'
+        required: true
+      STARTING_POINT:
+        description: 'Starting Point (e.g., a contractId, a contract address, ect.). Leave empty to start from the beginning or from last saved starting point.'
+        required: false
+
+jobs:
+  index-and-update:
+    name: Index ERC Contracts on Hedera ${{ github.event.inputs.HEDERA_NETWORK }} and Update Registry
+    runs-on: smart-contracts-linux-large
+    env:
+      INDEXER_PATH: ./tools/erc-repository-indexer/erc-contract-indexer
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.2
+        with:
+          egress-policy: audit
+
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: recursive
+
+      - name: Use Node.js [20]
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20
+
+      - name: Set Environment Variables
+        run: |
+          echo "HEDERA_NETWORK=${{ github.event.inputs.HEDERA_NETWORK }}" >> ${{ env.INDEXER_PATH }}/.env
+          echo "MIRROR_NODE_URL=${{ github.event.inputs.MIRROR_NODE_URL }}" >> ${{ env.INDEXER_PATH }}/.env
+          echo "STARTING_POINT=${{ github.event.inputs.STARTING_POINT }}" >> ${{ env.INDEXER_PATH }}/.env
+
+      - name: Install Dependencies
+        run: |
+          cd ${{env.INDEXER_PATH}}
+          npm install
+
+      - name: Backup Existing Registry
+        run: |
+          mkdir -p ${{ env.INDEXER_PATH }}/backup
+          if [ -d "${{ env.INDEXER_PATH }}/erc-registry/${{ github.event.inputs.HEDERA_NETWORK }}" ] && [ "$(ls -A ${{ env.INDEXER_PATH }}/erc-registry/${{ github.event.inputs.HEDERA_NETWORK }}/*.json 2>/dev/null)" ]; then
+            cp ${{ env.INDEXER_PATH }}/erc-registry/${{ github.event.inputs.HEDERA_NETWORK }}/*.json ${{ env.INDEXER_PATH }}/backup/
+            echo "Backup successful."
+          else
+            echo "No JSON files found to back up."
+          fi
+
+      - name: Start Timer
```

Comment:
If the CI steps on their own have a time I'm okay with just looking at that and we wouldn't have to calculate the time span

Reply:
Yeah I looked into that but GitHub doesn't seem to natively provide a built-in method to measure the duration of individual steps within a workflow. That's why I implemented two additional steps, start timer and stop timer. Well there was this pluggin https://github.com/marketplace/actions/github-workflow-duration-calculator but I thought  timer solution was simple enough.